### PR TITLE
connect 4 board. gotta start somewhere

### PR DESCRIPTION
**Pull request by deebloo**

### CODE REVIEW

File: shared/board.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Team } from './team';
+
+export class Board {
+  protected spaces: Team[] = new Array(this.rows * this.cols).fill(Team.Empty);
+
+  constructor(private rows: number, private cols: number) {}
+
+  get squares() {
+    return [...this.spaces];
```
Comment (zerobasedindex) [inline shared/board.ts:9]: what does this do? I know it's in the original, but I forgot.

File: xo/src/move-maker.ts [hunk 2]
```diff
@@ -41,16 +43,20 @@ export class MoveMaker {
     );
 
     let random = Math.floor(Math.random() * total) + 1; // Random inclusive between 1 and total
+    const length = counts.length;
     let move;
-    for (let i = 0; i < counts.length; i++) {
+
+    for (let i = 0; i < length; i++) {
```
Comment (zerobasedindex) [inline xo/src/move-maker.ts:49]: Does JS not optimize this?

File: xo/src/move-maker.ts [hunk 3]
```diff
@@ -41,16 +43,20 @@ export class MoveMaker {
...
       move = counts[i];
+
       if (move.count == 0) {
         continue;
       }
 
       if (random <= move.count) {
         return move;
       }
+
```
Comment (zerobasedindex) [inline xo/src/move-maker.ts:59]: You really like lots of white space

**Review by zerobasedindex — APPROVED** [review]

File: xo/src/move-maker.ts (hunk 2 above)
Comment (deebloo) [reply xo/src/move-maker.ts:49]: don't think so no.

File: shared/board.ts (hunk 1 above)
Comment (deebloo) [reply shared/board.ts:9]: it just makes sure that anything accessing the data from the outside doesn't accidentally manipulate the array

File: xo/src/move-maker.ts (hunk 3 above)
Comment (deebloo) [reply xo/src/move-maker.ts:59]: ahhhh can't help it. I do it as I read through stuff

Comment (deebloo) [reply xo/src/move-maker.ts:59]: and yes :)

**Pull request closed without merging**